index.rb

```diff
@@ -0,0 +1,139 @@
+# frozen_string_literal: true
+
+require 'yaml'
+require 'date'
+
+# Description/Explanation of Car class
+class Car
+  attr_accessor :cars, :make, :model, :year_from, :year_to, :price_from, :price_to, :date_or_price_sort,
+                :asc_desc_sort, :result_selection
+
+  def initialize(cars)
+    @cars = cars
+    @make = make
+    @model = model
+    @year_from = year_from
+    @year_to = year_to
+    @price_from = price_from
+    @price_to = price_to
+    @date_or_price_sort = date_or_price_sort
+    @asc_desc_sort = asc_desc_sort
+    @result_selection = @cars
+    @app_errors = []
+  end
+
+  def sanitize_text(text)
+    return if text.empty?
+
+    text.downcase.capitalize
+  end
+
+  def get_by_make(make_to_find)
+    return if make_to_find.length.zero?
+
+    res = @result_selection.filter { |obj| obj['make'] == sanitize_text(make_to_find) }
+    @result_selection = res.empty? ? @app_errors.push("Can't find by make: #{make_to_find}") : res
+  end
+
+  def get_by_model(model_to_find)
+    return if model_to_find.length.zero?
+
+    res = @result_selection.filter { |obj| obj['model'] == sanitize_text(model_to_find) }
+    @result_selection = res.empty? ? @app_errors.push("Can't find by model: #{model_to_find}") : res
+  end
+
+  def get_by_year_from(year_to_find)
+    return if year_to_find.length.zero?
+
+    res = @result_selection.filter { |obj| obj['year'] >= sanitize_text(year_to_find).to_i }
+    @result_selection = res.empty? ? @app_errors.push("Can't find by year: #{year_to_find}") : res
+  end
+
+  def get_by_year_to(year_to)
+    return if year_to.length.zero?
+
+    res = @result_selection.filter { |obj| obj['year'] <= sanitize_text(year_to).to_i }
+    @result_selection = res.empty? ? @app_errors.push("Can't find by year: #{year_to}") : res
+  end
+
+  def get_by_price_from(price)
+    return if price.length.zero?
+
+    res = @result_selection.filter { |obj| obj['price'] >= sanitize_text(price).to_i }
+    @result_selection = res.empty? ? @app_errors.push("Can't find by price_from: #{price}") : res
+  end
+
+  def get_by_price_to(price)
+    return if price.length.zero?
+
+    res = @result_selection.filter { |obj| obj['price'] <= sanitize_text(price).to_i }
+    @result_selection = res.empty? ? @app_errors.push("Can't find by price_to: #{price}") : res
+  end
+
+  def sort_by_criterias(direction, option = 'desc')
+    if 'price' == option
+      @result_selection = @result_selection.sort { |one, two| one[option] <=> two[option] }
+    elsif 'date_added' == option
+      @result_selection = @result_selection.sort { |one, two| one[option] <=> two[option] }
```

Comment:
I think sort by date works little bit wrong.
Try Date.strptime:
for example:
`@result_selection.sort_by { |car| Date.strptime(car[option], '%d/%m/%Y')`